device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceTwin/DeviceTwin.java

```diff
@@ -90,6 +94,14 @@ public IotHubMessageResult execute(Message message, Object callbackContext)
                         if (iotHubStatus == IotHubStatusCode.OK)
                         {
                             TwinState twinState = TwinState.createFromPropertiesJson(new String(dtMessage.getBytes(), Message.DEFAULT_IOTHUB_MESSAGE_CHARSET));
+
+                            if (dtMessage.getClientPropertiesCallback() != null)
+                            {
+                                ClientPropertiesCallback propCallBack = dtMessage.getClientPropertiesCallback();
+                                Object propCallBackContext = dtMessage.getClientPropertiesCallbackContext();
+                                propCallBack.execute(new ClientProperties(ClientPropertyCollection.fromMap(twinState.getReportedProperty()), ClientPropertyCollection.fromMap(twinState.getDesiredProperty())), propCallBackContext);
+                            }
```

Comment:
By default we will continue to execute other methods when a customer calls getTwin().

This means that we can have some unexpected doubling of callbacks if a customer implements both convention based and non-convention based operations.